File: frontend/lib/models/empresa.dart
class Empresa {
  final String id;
  final String nombre;
  final String nombreAdmin;
  final String emailAdmin;
  final int maxEmpleados;
  final DateTime fechaCreacion;

  Empresa({
    required this.id,
    required this.nombre,
    required this.nombreAdmin,
    required this.emailAdmin,
    required this.maxEmpleados,
    required this.fechaCreacion,
  });

  factory Empresa.fromJson(Map<String, dynamic> json) {
    return Empresa(
      id: json['id'],
      nombre: json['nombre'],
      nombreAdmin: json['nombre_admin'],
      emailAdmin: json['email_admin'],
      maxEmpleados: json['max_empleados'],
      fechaCreacion: DateTime.parse(json['fecha_creacion']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nombre': nombre,
      'nombre_admin': nombreAdmin,
      'email_admin': emailAdmin,
      'max_empleados': maxEmpleados,
      'fecha_creacion': fechaCreacion.toIso8601String(),
    };
  }
}

File: backend_db/app/models/empresa.py
from sqlalchemy import Column, String, Integer, TIMESTAMP
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.sql import func
from app.database import Base
import uuid

class Empresa(Base):
    __tablename__ = "empresas"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    nombre = Column(String, nullable=False)
    nombre_admin = Column(String, nullable=False)
    email_admin = Column(String, nullable=False, unique=True)
    plan = Column(String(30), nullable=False, default="starter")  
    max_empleados = Column(Integer, nullable=False)
    fecha_creacion = Column(TIMESTAMP, server_default=func.now())
